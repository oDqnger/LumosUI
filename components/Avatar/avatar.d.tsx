import { HTMLAttributes } from "react";

type SizeTypes = "sm" | "md" | "lg" | "xl" ;
type RadiusTypes = "none" | "sm" | "md" | "lg" | "full";
type ColorTypes = "default" | "primary" | "secondary" | "success" | "warning" | "danger" ;

interface AvatarProps extends HTMLAttributes<HTMLDivElement | HTMLImageElement> {
    src?: any,
    name?: string,
    className?: string,
}

interface StyledProps {
    size?: SizeTypes,
    radius?: RadiusTypes,
    color?: ColorTypes,
}

export type AllProps = AvatarProps & StyledProps;
