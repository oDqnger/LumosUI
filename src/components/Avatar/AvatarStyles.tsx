export const avatarColors = {
    "default": "border-slate-500",
    "primary": "border-sky-500",
    "secondary": "border-violet-600",
    "success": "border-green-500",
    "warning": "border-amber-600",
    "danger": "border-red-600",
}

export const radiusStyles = {
    "none": "rounded-none",
    "sm": "rounded-sm",
    "md": "rounded",
    "lg": "rounded-lg",
    "full": "rounded-full"
}

export const avatarSizesSrc = {
    "sm": "w-12 h-12",
    "md": "w-14 h-14",
    "lg": "w-16 h-16",
    "xl": "w-18 h-18"
}

export const avatarSizesName = {
    "sm": "p-1",
    "md": "p-2",
    "lg": "p-4",
    "xl": "p-8"
}

export const avatarNameBaseStyles = "font-semibold text-white bg-zinc-500 border-2 inline-block p-2";
