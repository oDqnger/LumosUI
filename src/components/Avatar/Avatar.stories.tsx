import { Meta, StoryObj } from "@storybook/react";
import Avatar from "./Avatar"

const meta: Meta<typeof Avatar> = {
    component: Avatar,
}

export default meta;
type Story = StoryObj<typeof Avatar>;

export const DefaultAvatar: Story = {
    render: (args) => <Avatar {...args} src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />,
    argTypes: {
        src: { control: 'text' },
        name: { control: 'text' },
        className: { control: 'text' },
        size: {
            control: 'select',
            options: ['sm', 'md', 'lg', 'xl']
        },
        radius: {
            control: 'select',
            options: ['none', 'sm', 'md', 'lg', 'full']
        },
        color: {
            control: 'select',
            options: ['default', 'primary', 'secondary', 'success', 'warning', 'danger']
        }
    }
}

// export const SmAvatar: Story = {
//     render: () => <Avatar size="sm" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const MdAvatar: Story = {
//     render: () => <Avatar size="md" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const LgAvatar: Story = {
//     render: () => <Avatar size="lg" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const XlAvatar: Story = {
//     render: () => <Avatar size="xl" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const RadiusNoneAvatar: Story = {
//     render: () => <Avatar radius="none" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const RadiusSmAvatar: Story = {
//     render: () => <Avatar radius="sm" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const RadiusMdAvatar: Story = {
//     render: () => <Avatar radius="md" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const RadiusLgAvatar: Story = {
//     render: () => <Avatar radius="lg" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const RadiusFullAvatar: Story = {
//     render: () => <Avatar radius="full" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const ColorDefaultAvatar: Story = {
//     render: () => <Avatar color="default" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const ColorPrimaryAvatar: Story = {
//     render: () => <Avatar color="primary" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const ColorSecondaryAvatar: Story = {
//     render: () => <Avatar color="secondary" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const ColorSuccessAvatar: Story = {
//     render: () => <Avatar color="success" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const ColorWarningAvatar: Story = {
//     render: () => <Avatar color="warning" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const ColorDangerAvatar: Story = {
//     render: () => <Avatar color="danger" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }

// export const NamedAvatar: Story = {
//     render: () => <Avatar name="Bob" />
// }

// export const CustomAvatar: Story = {
//     render: () => <Avatar className="border-2 border-teal-600 text-2xl w-32 h-32" src={"https://i.pravatar.cc/150?u=a042581f4e29026024d"} />
// }
