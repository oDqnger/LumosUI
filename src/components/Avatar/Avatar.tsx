import React, { forwardRef } from "react";
import { AllProps } from "./avatar.d";
import { avatarNameBaseStyles, avatarColors, avatarSizesSrc, avatarSizesName, radiusStyles } from "./AvatarStyles";

function Avatar(props: AllProps, ref) {
    
    const {
        src,
        name,
        size="md",
        radius="full",
        color="default",
        className,
    } = props;
    
    const combinedStylesSrc = `${className} ${radiusStyles[radius]} ${avatarSizesSrc[size]} ${avatarColors[color]} border-2`
    const combinedStylesName = `${className} ${avatarNameBaseStyles} ${radiusStyles[radius]} ${avatarSizesName[size]} ${avatarColors[color]}`

    if (src != null) {
        return (
            <>
                <img ref={ref} src={src} className={combinedStylesSrc} alt={name} />
            </>
        )
    } else {
        return (
            <>
                <div ref={ref} className={combinedStylesName}>
                    <span className="text-sm">{name?.substring(0, 3)}</span>
                </div><br />
            </>
        )
    }
}

export default forwardRef(Avatar);
