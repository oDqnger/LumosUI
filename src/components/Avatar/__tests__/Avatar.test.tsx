import '@testing-library/jest-dom/extend-expect';
import React from "react"
import { fireEvent, render } from "@testing-library/react"
import Avatar from "../Avatar"
import "@testing-library/jest-dom"

describe("Avatar component", () => {
    it("should match snapshot", () => {
        const { container } = render(<Avatar src={"https://i.pravatar.cc/150?u=a04258114e29026702d"} />)
        expect(container).toMatchSnapshot();
    })
    
    it("should render the component to the document", () => {
        const { container } = render(<Avatar />)
        expect(container).toBeInTheDocument();
    })
})
