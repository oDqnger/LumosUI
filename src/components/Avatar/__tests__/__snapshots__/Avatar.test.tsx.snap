// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should match snapshot 1`] = `
<div>
  <img
    class="undefined rounded-full w-14 h-14 border-slate-500 border-2"
    src="https://i.pravatar.cc/150?u=a04258114e29026702d"
  />
</div>
`;
